File: parsers/dart/test/stringify_test.dart
import 'package:synx/synx.dart';
import 'package:test/test.dart';

void main() {
  test('basic roundtrip', () {
    final r = parseFull('active true\nage 30\nname Wario\n');
    final out = stringify(r.root);
    expect(out, contains('name Wario'));
    expect(out, contains('age 30'));
    expect(out, contains('active true'));
  });

  test('multiline uses pipe', () {
    final o = SynxObject()..set('rules', synxString('a\nb\nc'));
    final out = stringify(synxObject(o));
    expect(out, contains('rules |'));
  });

  test('formatter sorts keys', () {
    final out = format('b 2\na 1\nc 3\n');
    final a = out.indexOf('a 1');
    final b = out.indexOf('b 2');
    expect(a, greaterThanOrEqualTo(0));
    expect(b, greaterThan(a));
  });

  test('formatter preserves directive', () {
    expect(format('!active\nname X\n'), startsWith('!active'));
  });
}

File: parsers/go/stringify_test.go
package synx

import (
	"strings"
	"testing"
)

func TestStringifyBasicRoundtrip(t *testing.T) {
	r := Parse("active true\nage 30\nname Wario\n")
	out := Stringify(r.Root)
	for _, s := range []string{"name Wario", "age 30", "active true"} {
		if !strings.Contains(out, s) {
			t.Fatalf("missing %q in %q", s, out)
		}
	}
}

func TestStringifyMultilineUsesPipe(t *testing.T) {
	o := NewObjectMap()
	o.Set("rules", String("a\nb\nc"))
	if !strings.Contains(Stringify(Object_(o)), "rules |") {
		t.Fatalf("multiline pipe missing")
	}
}

func TestFormatterSortsKeys(t *testing.T) {
	out := Format("b 2\na 1\nc 3\n")
	a := strings.Index(out, "a 1")
	b := strings.Index(out, "b 2")
	if a < 0 || b < 0 || a >= b {
		t.Fatalf("keys not sorted: %q", out)
	}
}

func TestFormatterPreservesDirective(t *testing.T) {
	out := Format("!active\nname X\n")
	if !strings.HasPrefix(out, "!active") {
		t.Fatalf("directive not preserved")
	}
}
